src/jit/lowerxarch.cpp

```diff
@@ -1373,25 +1373,10 @@ void Lowering::ContainCheckCallOperands(GenTreeCall* call)
 #endif // _TARGET_X86_
                 if (ctrlExpr->isIndir())
             {
-                MakeSrcContained(call, ctrlExpr);
                 // We may have cases where we have set a register target on the ctrlExpr, but if it
                 // contained we must clear it.
                 ctrlExpr->gtRegNum = REG_NA;
-            }
-        }
-    }
-    // If there is an explicit this pointer, we don't want that node to produce anything
-    // as it is redundant
-    if (call->gtCallObjp != nullptr)
```

Comment:
Can you explain why you are removing this code? I believe it's been there for a while, and I'm willing to believe that it is incorrect, but what led you to remove it?

Reply:
This code got it sight with the original problem, this code doesn't have any effects, because it set contain on node that do nothing and will be removed.
So the point is that it doesn't follow 'contained' logic that we have, because it doesn't have a user.

Reply:
I'm afraid I still don't understand the issue. Is it the case that:
- The node has already been removed, in which case it should not be encountered.
- The node is a "non-LIR" node (e.g. `ARGPLACE`)

Or is it something else?

Reply:
Yes, you are right, the node under this condition was 'GT_ARGPLACE' that is non-LIR node.

Reply:
Thanks for clarifying!